viam-cartographer.go

```diff
@@ -143,7 +143,10 @@ func initSensorProcess(cancelCtx context.Context, cartoSvc *CartographerService)
 	cartoSvc.sensorProcessWorkers.Add(1)
 	go func() {
 		defer cartoSvc.sensorProcessWorkers.Done()
-		sensorprocess.Start(cancelCtx, spConfig)
+		if jobDone := sensorprocess.Start(cancelCtx, spConfig); jobDone {
+			cartoSvc.jobDone = true
```

Comment:
Lets make `jobDone` an atomic bool, so that we don't need to worry bout data races between setting this & reading it from DoCommand.

Reply:
https://pkg.go.dev/sync/atomic#Bool